pyvrp/accept/MovingAverageThreshold.py

```diff
@@ -0,0 +1,127 @@
+from time import perf_counter
+
+import numpy as np
+
+
+class MovingAverageThreshold:
+    """
+    The Moving Average Threshold (MAT) criterion of [1]_. This criterion
+    accepts a candidate solution if it is better than a threshold value that is
+    based on the moving average of the objective values of recently observed
+    candidate solutions. The threshold is computed as:
+
+    .. math::
+
+       f(s^*) + \\tilde{\\eta} \\times \\left(
+          \\sum_{j = 1}^\\gamma \\frac{f(s^j)}{\\gamma} - f(s^*)
+       \\right)
+
+    where :math:`s^*` is the best solution observed in the last :math:`\\gamma`
+    iterations, :math:`f(\\cdot)` indicates the objective function,
+    :math:`\\gamma \\in \\mathbb{N}` is a parameter, and each :math:`s^j` is a
+    recently observed solution. The recently observed solutions are stored in
+    a ``history`` attributed of size at most :math:`\\gamma`.
+
+    The algorithm incorporates a decreasing factor :math:`\\tilde{\\eta}` that
+    converges to zero as the search approaches its maximum time limit or
+    iterations. It is calculated as:
+
+    .. math::
+        \\tilde{\\eta} = \\eta \\times \\min\\left( 1 - \\frac{t}{T},
+            1 - \\frac{i}{I} \\right)
+
+    where :math:`\\eta \\ge 0` is a parameter, :math:`t` is the time elapsed
+    since the start of the algorithm, :math:`T` is the imposed maximum runtime,
+    :math:`i` is the number of iterations performed, and :math:`I` is the
+    maximum number of iterations allowed. The most restrictive limit between
+    the two is used to compute the factor.
+
+    Parameters
+    ----------
+    eta
+        Used to determine the threshold value. Larger values of :math:`\\eta`
+        result in more accepted candidate solutions. Must be non-negative.
+    gamma
+        History size. Must be positive.
+    max_runtime
+        Maximum runtime in seconds. As the search approaches this time limit,
+        :math:`\\tilde{\\eta} \\to 0`. Must be non-negative. Default is
+        ``float('inf')``, meaning that :math:`\\tilde{\\eta}` stays constant.
+    max_iterations
+        Maximum number of iterations. As the search approaches this limit,
+        :math:`\\tilde{\\eta} \\to 0`. Must be non-negative. Default is
+        ``float('inf')``, meaning that :math:`\\tilde{\\eta}` stays constant.
+
+    References
+    ----------
+    .. [1] Máximo, V.R. and M.C.V. Nascimento. 2021. A hybrid adaptive iterated
+           local search with diversification control to the capacitated vehicle
+           routing problem, *European Journal of Operational Research* 294 (3):
+           1108 - 1119.
+    """
+
+    def __init__(
+        self,
+        eta: float,
+        gamma: int,
+        max_runtime: float = float("inf"),
+        max_iterations: float = float("inf"),
```

Comment:
This is inherently integral; why a float?

Reply:
Yea that's right, I wanted to make use of `float('inf')` like the CLI, so I type-hinted it with `float`. I think rewriting with None's should also do the trick.